src/utils/actions.js

```diff
@@ -82,19 +84,44 @@ const baseIgActions = {
       }
     }
 
-    const boolMap = action.condition.map(condition => {
-      if (condition.expression) {
-        const expression = condition.expression;
-        return evaluateExpression(expression, resources, variables);
-      } else {
-        // default true for non-applicability conditions
-        return true;
-      }
-    });
+    const boolMap = await Promise.all(
+      action.condition.map(async condition => {
+        if (condition.expression) {
+          const { expression, language } = condition.expression;
+
+          if (language === 'text/fhirpath') {
+            return isTrue(fhirpath.evaluate(resources, expression, variables));
+          } else if (language === 'text/cql') {
+            // Spec allows for multiple libraries, how can we tell which one is the main?
+            // Intentionally simplified and assuming the first library is the main
+            let library;
+            const planDefLib = context.planDefinition.library;
+            if (planDefLib?.length > 0) {
+              const libraryId = planDefLib[0];
+              library = db.select(LIBRARYS, l => l.resource.id === libraryId)[0];
+            }
+
+            if (!library) {
+              debug(`Library not defined in PlanDefinition or not found in database.`);
+              return false;
+            }
+
+            return evaluateCQL(
+              createBundle(resources, 'content'),
+              expression,
+              library,
+              context.patient.id
+            );
```

Comment:
Unless we care about the debug on line 105 this can be simplified to
```suggestion
            const planDefLib = context.planDefinition.library;
            if (planDefLib?.length > 0) {
              const libraryId = planDefLib[0];
              const library = db.select(LIBRARYS, l => l.resource.id === libraryId)[0];
              if (library) {
                 return evaluateCQL(
                  createBundle(resources, 'content'),
                  expression,
                  library,
                  context.patient.id
                );
              }
            }
            return false;
```